components/style/values.rs

```diff
@@ -1477,6 +1477,43 @@ pub mod specified {
     }
 
     #[derive(Clone, Copy, Debug, PartialEq, PartialOrd, HeapSizeOf)]
+    pub struct Number(pub CSSFloat);
+
+    impl Number {
+        pub fn parse(input: &mut Parser) -> Result<Number, ()> {
+            parse_number(input).map(Number)
+        }
+
+        fn parse_with_minimum(input: &mut Parser, min: CSSFloat) -> Result<Number, ()> {
+            match parse_number(input) {
+                Ok(value) if value < min => Err(()),
+                value => value.map(Number),
```

Comment:
Using a newtype as a function in this way seems a bit weird, but maybe it's a totally acceptable pattern. Can you verify that this is acceptable Rust? (No need to block landing on it)